api/controllers/api-labels-post.js

```diff
@@ -0,0 +1,60 @@
+module.exports = {
+  friendlyName: 'Add Label',
+  description: 'Add a new label to the database',
+  inputs: {
+    id : {
+      type: "string"
+    },
+    name: {
+      type: 'string',
+      required: true
+    },
+    hotKey: {
+      type: 'string',
+    },
+    source: {
+      type: 'string',
+    }
+  },
+  exits: {
+    success: {
+      statusCode: 200,
+      description: 'The session was created.'
+    },
+    validationFailed: {
+      statusCode: 400,
+      description: 'There was an error processing the request, see JSON attached.'
+    },
+    forbidden: {
+      statusCode: 403,
+      description: 'The user is not allowed to make that request.'
+    },
+  },
+
+  fn: async function (inputs, exits) {
+
+    let errorsObject = {};
+
+    // if (!this.req.session.userId) {
+    //   return exits.forbidden();
+    // }
+
+    let matchingName = await Labels.find({where: { name: inputs.name.trim().toLowerCase() }, });
+    let newLabel = {};
+    if(matchingName.length){
+      newLabel = await Labels.update({ name: inputs.name }).set({ name: inputs.name.trim().toLowerCase(), hotKey: inputs.hotKey, source: inputs.source }).fetch();
```

Comment:
i am curious if you tested this `update()` and `create()` when they will _fail_. (the only thing i can think of that comes to mind is by using a **duplicate name** i guess?)  i _suspect_ (hope!) the db-action will fail, due to the `unique: true` constraint.  i am just wondering how it looks to the user?

there _might_ need to be some sort of "catch" here to cushion that.  or (even better!) it will just work and shows the user an error to the effect of "duplicate name".  :crossed_fingers:

Reply:
Thank you for the suggestions. I adjusted the label.name constraints orders. Now first db will check if the name is empty, if true return error message. Then it will check if this name already exists, if matchingName.length is true, it means there is existing label with the same name, then we need to update it. If not, it means there is no such label yet, so we will create one. So if user inputs a name, either it's duplicate or not, db will have a way to deal with it.